### ISSUE TITLE
Error 404 deploying to ios device after xcode upgraded to 8.3.2

### ISSUE THREAD

**Issue by BigFoppa** [Microsoft/remotebuild#31]
As I have described in my comments to issue #26 I can not deploy to Remote device from Visual Studio 2015. I am able to build after the latest changes to remotebuild but not deploy.  

My current setup:
PC: VS2015, Cordova 6.4.0, ios 4.3.1
Mac: macOS 10.12.4, xcode 8.3.2 and cordova 6.4.0 globally installed.

This is the message after debug build to Remote device:

![image](https://cloud.githubusercontent.com/assets/9096750/25905124/6f7a4e94-35a1-11e7-8567-7a98ff74a989.png)

I believe I have tried everything that is suggested around this kind of issue.

**Comment by domisginger** [comment]
I also get this issue. I can run a debug build to 'Local Device' but with 'Remote Device' I get 404

`
------ Downloading completed build files from https://SERVERNAME.uk:3000/cordova/build/6638/download to C:\TFSONLINE\SpeMobile\SpeMobile\bld\iOS\Debug
------ Done downloading the build outputs to C:\TFSONLINE\SpeMobile\SpeMobile\bld\iOS\Debug
------ Extracting the build outputs to C:\TFSONLINE\SpeMobile\SpeMobile\bin\iOS\Debug
------ Done extracting the build outputs to C:\TFSONLINE\SpeMobile\SpeMobile\bin\iOS\Debug
------ Copying back to project: ios
1>Done building project "SpeMobile.jsproj".
2>------ Deploy started: Project: SpeMobile, Configuration: Debug iOS ------
2>Requesting deploy on remote iOS device for buildNumber 6638 on server https://SERVERNAME.uk:3000/cordova...
2>https://SERVERNAME.uk:3000/cordova/build/6638/deploy The remote server returned an error: (404) Not Found.
`

If i call the deploy address in chrome back on windows machine I get

`
{
stdout: "Copying '/Users/adt/.taco_home/remote-builds/taco-remote/builds/6638/cordovaApp/platforms/ios/build/device/SpeMobile UAT.ipa' to device... DONE. Installing 'ionic.adt.spemobile' - CreatingStagingDirectory (5%) - ExtractingPackage (15%) - InspectingPackage (20%) - TakingInstallLock (20%) - PreflightingApplication (30%) - InstallingEmbeddedProfile (30%) - VerifyingApplication (40%) - CreatingContainer (50%) - InstallingApplication (60%) - PostflightingApplication (70%) - SandboxingApplication (80%) - GeneratingApplicationMap (90%) ",
stderr: "WARNING: could not locate iTunesMetadata.plist in archive! WARNING: could not locate Payload/SpeMobile UAT.app/SC_Info/SpeMobile UAT.sinf in archive! ",
code: null,
message: "Error installing app to device: WARNING: could not locate iTunesMetadata.plist in archive! WARNING: could not locate Payload/SpeMobile UAT.app/SC_Info/SpeMobile UAT.sinf in archive! "
}
`

**Comment by BigFoppa** [comment]
@domisginger thanks for that. I thought I was alone on this one :-)

**Comment by jlyshoel** [comment]
@BigFoppa I've also got the same issue. I'll be looking more in dept today.

**Comment by jlyshoel** [comment]
The deploy 404 starts when I run "brew upgrade libimobiledevice --HEAD" in accordance with this:
https://github.com/Microsoft/remotebuild/issues/5

The short answer seems to be that when not using the --HEAD for libimobiledevice remotebuild is able to run /deploy and return HTTP 200.  When using --HEAD for libimobiledevice, remotebuild returns 404 on deploy.

The strange thing about this is that the app actually deploys when remotebuild returns 404, but doesn't deploy when it returns 200.  This is deployment to physical devices. 

I get some warnings;
```
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/<appname>.app/SC_Info/<appname>.sinf in archive!
```

That's how far I've gotten.

**Comment by jlyshoel** [comment]
I tried running `cordova run --device` in the cordovaApp folder, and that starts and runs the app on the phone

**Comment by jlyshoel** [comment]
@BigFoppa Have a look at this, it's the same problem I'm having. I tried deployment using ideviceinstall and it gives a seg fault 11 as described: https://github.com/libimobiledevice/ideviceinstaller/issues/70

**Comment by BigFoppa** [comment]
@jlyshoel Yes, I agree, that  seems to be the same problem. Thanks for the hint.

**Comment by BigFoppa** [comment]
@AnBucyk this may not be your problem. But if it is please have a look at the issue reported here: : libimobiledevice/ideviceinstaller#70

**Comment by drewgillies** [comment]
Closing this as a dupe of https://github.com/Microsoft/remotebuild/issues/26. If this issue persists after 26 is fixed, feel free to reopen.

**Comment by BigFoppa** [comment]
@drewgillies I was in issue #26 explicitly asked to create a new issue for this problem. It should stay open as I see it.

**Comment by BigFoppa** [comment]
@drewgillies I can not see that I can reopen this issue.

**Comment by Ashley9** [comment]
I've also got the same issue...

PC: VisualStudio2015, node4.8.3, npm2.15.11, cordova6.5.0
Mac: Xcode8.3.2, node7.4.0, npm4.0.5, cordova6.5.0, remotebuild2.1.3, ios-deploy1.9.1

Could you give me some advice?

**Comment by BigFoppa** [comment]
@Ashley9 I am building and deploying to Local device instead of remote. Had problems locating the DeviceSupport folder on the Mac, the path in the documentation did not exist. Searched the mac for DeviceSupport and copied the files to the pc. That worked. But debugging don't. 
If you need to debug then you can use Xcode.
You can also use Cordova to deploy and run.

**Comment by fedlerner** [comment]
I'm having the same issue....
Is there a solution for this ?

**Comment by ekambarrao** [comment]
Hi, build successful but failing to deploy to IOS device. it is bringing up itunes and then shows the app there and device is identified but says error occured continue or not? and fails.

one error result states, version mismatch other says device may not be connected and awake.

I am using ipod 9.3 version ios and another one iphone 5s with 10.2.1 ios.

I am building apache cordova app in visual studio 2017 with macincloud remotebuild.

how to resolve this!? any hint is a big help!!

**Comment by Ashley9** [comment]
@BigFoppa I'm sorry for the late reply.
I still got an error, but I could debug. Thank you!

**Comment by fedlerner** [comment]
Does remotebuild support XCode 8.3.2 ? To compile and publish an app ?
Has this issue been fixed?

**Comment by Webrusterkk** [comment]
@fedlerner  remotebuild Supports Xcode 8.3.2 but  to publish an app , you need to copy the `ios` folder from the platforms and move to the MAC and from there you need to Publish the app to the store

**Comment by dmitriy-petrishin** [comment]
Hi guys, do you still have the same issue?
@ekambarrao @Ashley9 @BigFoppa Have you ever tried to use visual studio 2017 for deploying and debugging?

**Comment by Sroose** [comment]
Hello, issue is still present. (I'm using VS 2015 community edition)

**Comment by zzzxtreme** [comment]
hello. same problem here. is there no solution at all?

**Comment by zzzxtreme** [comment]
I changed to an ipad, and it works 👍

**Comment by fedlerner** [comment]
@Webrusterkk is there a tutorial online on how to proceed ?

**Comment by Webrusterkk** [comment]
@fedlerner unfortunately there are not any usefull online tutorial at one place, for start up i would recommend to start from the [VS Docs ](https://taco.visualstudio.com/en-us/docs/deploy-and-run-app/)
